### ISSUE TITLE
tutorial on integral image

### ISSUE THREAD

**Issue by johnnychen94** [JuliaImages/juliaimages.github.io#231]
I believe we need an `IntegralArrays` version of https://juliaimages.org/stable/examples/color_channels/indexed_image/#demo_indexed_image

**Comment by timholy** [comment]
That's an `IndirectArray` tutorial, not an `IntegralArray` tutorial. I don't see any tutorials that use the (soon to be deprecated) `integral_image`, am I missing something?

**Comment by johnnychen94** [comment]
Oh sorry for the inaccuracies. I notice the examples in https://github.com/JuliaImages/IntegralArrays.jl is pretty thin, I think we need a more insightful explanation on why we need this type (e.g., performance), and how it is used in image processing just like the IndirectArray demo.

**Comment by timholy** [comment]
Yeah, it's not a widely used type (so far, anyway). I once knew applications that needed it, but I kind of forget them now. Looks like https://www.codeproject.com/Articles/441226/Haar-feature-Object-Detection-in-Csharp has a practical use-case.

**Comment by johnnychen94** [comment]
https://github.com/JuliaImages/ImageFiltering.jl/issues/241 is a good use case; we can also use this as a demo.
